src/interpreter/mod.rs

```diff
@@ -595,7 +595,15 @@ where
                         return res;
                     }
                 }
-                Terminal::PkH(ref pkh) => {
+                Terminal::PkH(ref pk) => {
+                    debug_assert_eq!(node_state.n_evaluated, 0);
+                    debug_assert_eq!(node_state.n_satisfied, 0);
+                    let res = self.stack.evaluate_pk(&mut self.verify_sig, pk);
```

Comment:
The evaluation would still be evaluate_pkh. We would pass pk.to_pubkeyhash() as an arg

Reply:
The following error occurs in this case:
```bash
error[E0716]: temporary value dropped while borrowed
   --> src/interpreter/mod.rs:601:78
    |
557 | impl<'intp, 'txin: 'intp> Iter<'intp, 'txin>
    |      ----- lifetime `'intp` defined here
...
601 |                     let res = self.stack.evaluate_pkh(&mut self.verify_sig, &pk.to_pubkeyhash());
    |                               -----------------------------------------------^^^^^^^^^^^^^^^^^^-- temporary value is freed at the end of this statement
    |                               |                                              |
    |                               |                                              creates a temporary which is freed while still in use
    |                               argument requires that borrow lasts for `'intp`
```
Here, we'd need to either create the hash outliving the scope (since `'intp` is the liftime required for it's reference) or change some of our current APIs. If there's any solution, kindly let me know!